src/components/Layout/Header/NavBar/KeepKey/AwaitKeepKey.tsx

```diff
@@ -0,0 +1,52 @@
+import { InfoIcon } from '@chakra-ui/icons'
+import { FlexProps } from '@chakra-ui/layout'
+import { Box, Button, Divider, Flex, useColorModeValue } from '@chakra-ui/react'
+import Polyglot from 'node-polyglot'
+import React from 'react'
+import { useTranslate } from 'react-polyglot'
+import { Text } from 'components/Text'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+export type AwaitKeepKeyProps = {
+  translation: string | null | [string, number | Polyglot.InterpolationOptions]
+  children?: React.ReactNode
+  onCancel?: () => Promise<void> | void
+} & FlexProps
+
+export const AwaitKeepKey = ({ children, translation, onCancel, ...props }: AwaitKeepKeyProps) => {
+  const translate = useTranslate()
+  const {
+    setAwaitingDeviceInteraction,
+    state: { awaitingDeviceInteraction, wallet },
+    load
+  } = useWallet()
+  const blueShade = useColorModeValue('blue.500', 'blue.200')
+
+  const cancel = async () => {
+    if (onCancel) {
+      await onCancel()
+    }
+    setAwaitingDeviceInteraction(false)
+    await wallet?.cancel()
+    await load()
+  }
+
+  return awaitingDeviceInteraction ? (
+    <>
+      <Divider />
+      <Box p={3}>
+        <Flex {...props}>
+          <InfoIcon color={blueShade} mt={1} />
+          <Box ml={3}>
+            <Text translation={translation} mb={3} fontWeight='medium' color={blueShade} />
+            <Button colorScheme='blue' variant='ghost-filled' onClick={cancel} size='sm'>
+              {translate('common.cancel')}
```

Comment:
does `Button` not take a `t` prop to pass in a translation path string?

Reply:
It doesn't, but I do see value in adding our own button wrapper that handles this, much like we've already done for `Text`.